include/logging/log_core.h

```diff
@@ -15,6 +15,10 @@
 extern "C" {
 #endif
 
+#if UINTPTR_MAX == 0xFFFFFFFFFFFFFFFFUL
```

Comment:
We also have BUILD_ASSERT_MSG() macro that could be used here, so something like this
```
BUILD_ASSERT_MSG(sizeof(void *) == 4, "Logger does not support 64 bit architecture");
```
You seem to be using it already in log_msg.h. I have no preference for this so this proposal is ok too.